### ISSUE TITLE
Frost 579 fails to create a new backup

### ISSUE THREAD

**Issue by bdb** [X-Ryl669/Frost#15]
Trying to setup a new backup:

```
$ ls -l ~/Desktop/test
total 136
-rw-r--r--  1 brian  501  66383 Dec  8 14:41 raven.txt
$ ls -l ~/Desktop/test2
ls: /Users/brian/Desktop/test2: No such file or directory
$ ./build/Frost -vv --backup ~/Desktop/test --remote ~/Desktop/test2
error: Bad argument for backup, index path does not exists

Frost (C) Copyright 2017 - Cyril RUSSO (This software is BSD licensed) 
Frost is a tool used to efficiently backup and restore files to/from a remote
place with no control other the remote server software.
No warranty of any kind is provided for the use of this software.
Current version: 3 build 579. 
...
```

Key store is empty:
```
$ ls ~/.frost/

```

**Comment by X-Ryl669** [comment]
For the next days, please add `--index ~/Desktop/test` to the command line.
With the new encrypted index feature, it's required to set the path to the clear-text index's directory.

I agree, this is not convenient if you don't use/need it, so I'll sort this out when `--safeindex` is not given to revert to the previous behavior of writing the index in the remote's folder.

**Comment by bdb** [comment]
OK, thanks. I figured something was missing, but I couldn't figure out what from the built-in help nor the web doc.

**Comment by bdb** [comment]
OK, backup starts now, but it fails. I thought because this index was in the src folder and it tried to backup the index, but moving the index to `~/Desktop` still fails:

```
Backup: Analysing:  [1/3]   Mismatch in metadata P1000004/20021F137/41ED/80/2/1F5/1F5/5A3194CF/5A3194CF/5A3194CF vs P1000004/20021F137/41ED/80/4/1F5/1F5/5A3194CF/5A3194CF/5A3194CF
Backup: Analysing: index.frost [2/3]Mismatch in metadata P0/0/81A4/78/1/1F5/1F5/5A3194CF/5A3194CF/5A3194CF vs P1000004/200224285/81A4/78/1/1F5/1F5/5A3194CF/5A3194CF/5A3194CF
Backup: index.frost [2/3]           
Backup: Analysing: raven.txt [3/3]Mismatch in metadata P0/0/81A4/1034F/1/1F5/1F5/5A317C5E/5A2AF901/5A317C5F vs P1000004/20021F13D/81A4/1034F/1/1F5/1F5/5A317C5E/5A2AF901/5A317C5F
Backup: raven.txt [3/3]                            
Can't backup the test folder: Can't close the last multichunk
```

The backup folder `test2` is not created.

It says mismatch in metadata, but I don't see any mismatch in the fields.

**Comment by X-Ryl669** [comment]
Strange error. I never tried to backup the folder containing the index file.

Can you run the tests `./Frost --test roundtrip` to see if it fails ?

I'll install clang4 to get the exact warning test you had, so I can figure out if the warning were actually errors.

**Comment by X-Ryl669** [comment]
I've just seen your edit. It should create the remote if it does not exists, so it's definitively a bug. 
Not sure I can handle backing up the index file however, since the index is memory mapped, so it's being changed while the backup progresses and this might confuse the software.
I'll also add a check for this to prevent this case.

I know I'll have to improve the online documentation, it's in my todo list ;-)

**Comment by X-Ryl669** [comment]
I think this is fixed with commit `f467712`. Feel free to comment if it's not.
